### PR TITLE
[GAM] NaN MOJO fix + tests + cleanup

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

**Comment by michalkurka** [comment]
@Pscheidl please take over this PR while @honzasterba is on vacation

**Comment by Pscheidl** [comment]
@michalkurka @honzasterba Got to it now.

**Comment by Pscheidl** [comment]
Rebased on top of rel-zeno to get the latest fixes in.

File: h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java [hunk 1]
```diff
@@ -83,8 +81,8 @@ private void imputeMissingWithMeans(double[] data) {
       if (Double.isNaN(data[ind])) data[ind] = _catNAFills[ind];
 
     if (data.length == nfeatures()) { // using centered gam cols, nfeatures denotes centered gam columns
-      for (int ind = 0; ind < _numsCenter; ind++)
-        if (Double.isNaN(data[ind + _cats])) data[ind] = _numNAFillsCenter[ind];
+      for (int ind = _cats; ind < _numsCenter + _cats; ind++)
```
Comment (Pscheidl) [inline h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java:84]: This is the actual **fix** for the "NaN problem". The imputation of categoricals has been done in a correct way, but the imputation of the rest was started using bad index - it overrode existing categoricall imputation and did not reach the end. Therefore, the fix is only in indexing.

I took a look at how `DataInfo.extractDenseRow` does it (actually used for GAM native model scoring) - looks the same to me.

cc: @wendycwong @michalkurka @honzasterba

**Comment by Pscheidl** [comment]
Please review @wendycwong and (I guess semi-optionally) @michalkurka.

**Comment by Pscheidl** [comment]
Tests look good, updated on top of latest rel-zeno because of R CRAN check fixes before merging.

**Pull request merged**